lua/neo-tree/sources/common/file-nesting.lua

```diff
@@ -179,24 +199,39 @@ function M.nest_items(context)
   if M.is_enabled() == false or table_is_empty(context.nesting) then
     return
   end
+
+  -- First collect all nesting relationships
+  local all_nesting_relationships = {}
   for _, config in pairs(context.nesting) do
     local files = config.nesting_callback(config, context.all_items)
-    local folder = context.folders[config.parent_path]
-    for _, to_be_nested in ipairs(files) do
-      table.insert(config.children, to_be_nested)
-      to_be_nested.is_nested = true
-      to_be_nested.nesting_parent = config
-      if folder ~= nil then
-        for index, file_to_check in ipairs(folder.children) do
-          if file_to_check.id == to_be_nested.id then
-            table.remove(folder.children, index)
+    if files and #files > 0 then
+      table.insert(all_nesting_relationships, {
+        parent = config,
+        children = files,
+      })
+    end
+  end
+
+  -- Then apply them in order
```

Comment:
if i'm not mistaken, the iteration order of `pairs` is non-deterministic in Lua/LuaJIT, so  there isn't really an order here. As an example, using your nesting rules plugin (nice btw) with something like this directory structure:

![image](https://github.com/user-attachments/assets/afcc76cd-3d3d-4d8f-9a65-7929730d479f)

the file that holds babelrc isn't the same across nvim launches.

It's not too important, this PR is still an improvement over the previous behavior so I'm down to merge this as-is. As a temporary fix, we could iterate over context.nesting using `vim.spairs` (which will iterate in order of the keys, i.e. the absolute filepaths), but I think the easiest route would be a `priority` field for the nesting rules? what would you prefer?
```suggestion
  -- Then apply them (TODO: Determine how to order the rules)
```